### ISSUE TITLE
Run ptx directly in GPGPUSim

### ISSUE THREAD

**Issue by ishitachaturvedi** [gpgpu-sim/gpgpu-sim_distribution#272]
Can I run ptx using gpgpusim, instead of running a binary?

**Comment by topcuburak** [comment]
@ishitachaturvedi 
Yes, you can. I suggest you read simulator papers. The last one (Accel-Sim paper) mentioned that the accel-sim version could execute the ptx code. However, they suggested using the SASS instead of the ptx because the ptx is a virtual machine code and executes with some assumptions. In that paper, they also provided correlation results explaining the difference between SASS and PTX execution results concerning real hardware execution.

**Comment by nothingface0** [comment]
@ishitachaturvedi 

Assuming you want a PTX of a specific version (let's say 7.0), you need to compile your binary with `-gencode=arch=compute_70,code=compute_70`. This will embed PTX 7.0 in your executable (`code=` determines what will end up inside the executable, `compute_XX` determines PTX instructions). 

Then, doing a `source setup_environment` should be enough to let you execute the binary via GPGPU-Sim.
